src/main/java/de/tum/in/www1/artemis/service/plagiarism/ProgrammingPlagiarismDetectionService.java

```diff
@@ -326,8 +326,10 @@ private Language getJPlagProgrammingLanguage(ProgrammingExercise programmingExer
      * @param minimumScore        consider only submissions whose score is greater or equal to this value
      * @return an unmodifiable list containing the latest text submission for every participation
      */
-    public List<ProgrammingExerciseParticipation> filterStudentParticipationsForComparison(ProgrammingExercise programmingExercise, int minimumScore) {
+    public List<ProgrammingExerciseParticipation> findStudentParticipationsForComparison(ProgrammingExercise programmingExercise, int minimumScore) {
+        long start = System.nanoTime();
         var studentParticipations = studentParticipationRepository.findAllForPlagiarism(programmingExercise.getId());
+        log.info("findAllForPlagiarism took {}", TimeLogUtil.formatDurationFrom(start));
 
         return studentParticipations.parallelStream().filter(participation -> !participation.isPracticeMode())
```

Comment:
You could also easily include the filtering for the practice mode and the minimum score in the query. This should decrease the retrieved information from the data base even more

Reply:
p.testRun = FALSE is already included.

The other one is more tricky, because it is optional. Let's keep it simple for now. This PR is not about the best performance ever, it's really about fixing a very very slow query with already ~98% improvement. Let's stay with the Pareto and KISS principles and avoid adding too complex and error prone queries